sim/midas/src/main/cc/bridges/test_finisher.cc

```diff
@@ -0,0 +1,37 @@
+#ifdef TESTFINISHERBRIDGEMODULE_struct_guard
+
+#include "test_finisher.h"
+#include <iostream>
+
+test_finisher_t::test_finisher_t(
+	simif_t* sim, 
+  TESTFINISHERBRIDGEMODULE_struct * mmio_addrs,
+	unsigned int num_messages,
+  unsigned int* isErr,
+	const char* const* msgs): 
+	bridge_driver_t(sim), 
+	mmio_addrs(mmio_addrs), 
+	isErr(isErr),
+	msgs(msgs)
+{
+  this->mmio_addrs = mmio_addrs;
+}
+
+test_finisher_t::~test_finisher_t() {
+	free(this->mmio_addrs);
+}
+
+void test_finisher_t::tick() {  //reads the MMIOs every tick
+  if (read(this->mmio_addrs->out_status)) {
+	  msg_id = read(this->mmio_addrs->out_errCode);
+		if(this->isErr[msg_id]==0) {
+			fprintf(stderr, "*PASS* \n");
+		} else {
+			fprintf(stderr, "*FAIL* \n");
+		}
+	  fprintf(stderr, "Test completed with message:  ", msg_id);
```

Comment:
Use (favor) the c++ std iostreams consistently.

Reply:
Yup.